src/opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.cpp

```diff
@@ -449,19 +456,29 @@ namespace Opm {
     static std::vector< double > findKrgr( const TableManager* tm ) {
         const auto num_tables = tm->getTabdims().getNumSatTables();
         const auto& sgofTables = tm->getSgofTables();
+        const auto& slgofTables = tm->getSlgofTables();
         const auto& sgfnTables = tm->getSgfnTables();
 
-        const auto& famI = [&sgofTables]( int i ) {
+        const auto& famI_sgof = [&sgofTables]( int i ) {
             return sgofTables.getTable< SgofTable >( i ).getKrgColumn().front();
         };
 
+        const auto& famI_slgof = [&slgofTables]( int i ) {
+            return slgofTables.getTable< SlgofTable >( i ).getKrgColumn().back();
+        };
+
         const auto& famII = [&sgfnTables]( int i ) {
             return sgfnTables.getTable< SgfnTable >( i ).getKrgColumn().back();
         };
 
         switch( getSaturationFunctionFamily( tm ) ) {
             case SatfuncFamily::I:
-                return fun::map( famI, fun::iota( num_tables ) );
+                if( sgofTables.empty() && slgofTables.empty() )
+                    throw std::runtime_error( "Saturation keyword family I requires either sgof or slgof non-empty" );
+                if( !sgofTables.empty() )
+                    return fun::map( famI_sgof, fun::iota( num_tables ) );
+                else
+                    return fun::map( famI_slgof, fun::iota( num_tables ) );
```

Comment:
Can probably use `std::iota` now?

Reply:
> Can probably use `std::iota` now?

Not really&mdash;at least not as succinctly.  Function `std::iota()` fills values into an existing sequence whereas `fun::iota()` creates a new sequence (represented as a `std::vector`).  Using `std::iota()` here would mean writing something along the lines of
```C++
auto seq = std::vector<int>(num_tables);
std::iota(seq.begin(), seq.end(), 0);
return fun::map(f, seq);
```